framework/src/auxkernels/NodalPatchRecovery.C

```diff
@@ -0,0 +1,224 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "NodalPatchRecovery.h"
+#include "SwapBackSentinel.h"
+#include "libmesh/utility.h"
+
+template <>
+InputParameters
+validParams<NodalPatchRecovery>()
+{
+  InputParameters params = validParams<AuxKernel>();
+  params.addParam<unsigned>("patch_polynomial_order", "polynomial order");
```

Comment:
Add a better description of this:
Polynomial order used in least squares fitting of material property over the local patch of elements connected to a given node